core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1366,16 +1370,17 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
     def state = ControllerState.IsrChange
 
     override def process(): Unit = {
+      val currentSequenceNumbers = zkUtils.getChildrenParentMayNotExist(ZkUtils.IsrChangeNotificationPath)
```

Comment:
We'd want to add a comment explaining why this is important. Similar to how we did for the other event where we applied a similar pattern.

Reply:
Sure. I added the following comment:

```
// Read the current isr change notification znodes from ZK again instead of using sequenceNumbers
// to increase the odds of processing recent isr changes in a single ControllerEvent
// and to reduce the odds of trying to access znodes that have already been deleted (KAFKA-5879).
```